fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java

```diff
@@ -113,6 +114,23 @@ public GroupExpression removeGroupExpression(GroupExpression groupExpression) {
         return groupExpression;
     }
 
+    /**
+     * remove all logical and physical expressions
+     * if this group is merged with other group, all the expressions should be removed.
+     */
+    public void removeAllExpressions() {
+        Iterator<GroupExpression> iter = logicalExpressions.iterator();
```

Comment:
i think we should not setOwnerGroup to null here. since in java all things is reference, we do add to destination group first and then to remove all expressions in source. if we do  setOwnerGroup to null here, the result will be wrong
BTW, maybe call list.clear is better.